src/lsp/cobol_ast/terms.ml

```diff
@@ -149,12 +149,27 @@ and expression =
 and _ cond =
   (* TODO: group generalized expressions together (class, sign, omitted) *)
   | Expr: expression -> [>simple_] cond (* exp (bool), ident (bool, cond, switch) *)
-  | Relation: expression * relop * expression -> [>simple_] cond (* general, bool, pointer *)
+  | Relation: binary_relation -> [>simple_] cond     (* general, bool, pointer *)
+  | Abbrev: abbrev_combined_relation -> [>simple_] cond (* abbreviated *)
   | ClassCond: expression * class_ -> [>simple_] cond (* exp = ident *)
   | SignCond: expression * signz -> [>simple_] cond (* exp = arith exp *)
   | Omitted: expression -> [>simple_] cond (* exp = ident *)
   | Not: _ cond -> [>complex_] cond
-  | Logop: _ cond * logop * _ cond -> [>complex_] cond (* TODO: move logop left *)
+  | Logop: _ cond * logop * _ cond -> [>complex_] cond
+
+and binary_relation =
+  expression * relop * expression
+
```

Comment:
Do you think it would be possible to add comments with the corresponding syntax in COBOL ? it's not always easy to match such types with what they encode, and reciprocally